weather_aplication/MainViewController.swift

```diff
@@ -1,9 +1,123 @@
+        dataArray.append(CellData(hour: "15:00", temperature: 24, pressure: 1015, windSpeed: 23))   //6
+        dataArray.append(CellData(hour: "16:00", temperature: 19, pressure: 1019, windSpeed: 25))   //7
+        dataArray.append(CellData(hour: "17:00", temperature: 25, pressure: 1017, windSpeed: 34))   //8
+        dataArray.append(CellData(hour: "18:00", temperature: 27, pressure: 1014, windSpeed: 32))   //9
+        dataArray.append(CellData(hour: "19:00", temperature: 30, pressure: 1009, windSpeed: 14))   //10
+        dataArray.append(CellData(hour: "20:00", temperature: 31, pressure: 1011, windSpeed: 15))   //11
+        dataArray.append(CellData(hour: "21:00", temperature: 32, pressure: 1012, windSpeed: 16))   //12
+    }
     
     override func viewDidLoad() {
+        super.viewDidLoad()
+        createDataArray()
+        setupView()
+        setupConstraints()
+        
+    }
+    
+    func setupView() {
         view.backgroundColor = .lightGray
+        
+        cityLabel.textAlignment = .center
+        cityLabel.text = cityString
+        view.addSubview(cityLabel)
+        
+        weatherTableView.dataSource = self
+        weatherTableView.delegate  = self
+        weatherTableView.backgroundColor = .white
+        view.addSubview(weatherTableView)
+        weatherTableView.register(CustomTableViewCell.self, forCellReuseIdentifier: "cell")
+    }
+    
+    func setupConstraints() {
+        
+        cityLabel.snp.makeConstraints { make in
+            make.top.equalTo(view.snp.topMargin)
+            make.leftMargin.equalToSuperview().offset(kSideMargin)
+            make.rightMargin.equalToSuperview().offset(-kSideMargin)
+            make.centerX.equalTo(view.snp.centerX)
+        }
+        
+        weatherTableView.snp.makeConstraints { make in
+            make.top.equalTo(cityLabel.snp.bottom).offset(kTopMargin)
+            make.leftMargin.rightMargin.bottomMargin.equalToSuperview()
+        }
+    }
+}
+
+extension HourlyWeatherViewController: UITableViewDelegate, UITableViewDataSource {
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+      return dataArray.count
+    }
+    
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        guard let cell = tableView.dequeueReusableCell(withIdentifier: "cell", for: indexPath as IndexPath) as? CustomTableViewCell else {fatalError("Unble to create cell")}
+        cell.hourLabel.text = dataArray[indexPath.row].hour
+        cell.temperatureLabel.text = "\(dataArray[indexPath.row].temperature)C"
+        cell.pressureLabel.text = "\(dataArray[indexPath.row].pressure)hPa"
+        cell.windSpeedLabel.text = "\(dataArray[indexPath.row].windSpeed)km/h"
+        return cell
+    }
+}
+
+struct CellData {
+    lazy var hour = String()
+    lazy var temperature = Int()
+    lazy var pressure = Int()
+    lazy var windSpeed = Int()
```

Comment:
Struktury nie powinny generalnie mieć defaultowej wartości, w inicie powinniśmy przekazywać wszystkie potrzebne informacje. Same dane powinny raczej być stałe, a nie zmienne, bo jak już je otrzymamy to nie mamy zamiaru ich zmieniać. Przeczytaj sobie też i napisz mi co robi przedrostek _lazy_ przy takiej deklaracji zmiennej.

Reply:
lazy sprawia że pamięć nie jest alokowana dopóki zmienna nie zostanie użyta

Reply:
Zgadza się, ale tak czy tak nie chcemy dawać tym zmiennym defaultowej wartości. Zamiast tego chcemy po prostu określić typ jaki ma mieć dana zmienna, tak żeby dopiero w momencie tworzenia instancji nadać jej wartość.